### PR TITLE
Fix: add price precision for data in table on landing page

### PR DESCRIPTION
**Pull request by dmitrydenisov183**
https://openware-inc.atlassian.net/browse/OPD-449

Added price_precision for table on the Landing page
image

### CODE REVIEW

File: web/src/components/TickerTable/index.tsx [hunk 1]
```diff
@@ -9,14 +9,18 @@ interface Props {
     markets: Market[];
     redirectToTrading: (key: string) => void;
     setCurrentBidUnit: (key: string) => void;
+    isFixedPrecision: boolean
```
Comment (josadcha) [inline web/src/components/TickerTable/index.tsx:12]: please remove this prop, there is no case when we need Market table without fixed precision.

File: web/src/components/TickerTable/index.tsx [hunk 2]
```diff
@@ -9,14 +9,18 @@ interface Props {
...
 }
 
+const FIXED_PRECISION = 2;
```
Comment (josadcha) [inline web/src/components/TickerTable/index.tsx:15]: name this as FIXED_VOL_PRECISION, move to src/constants.ts file and import here

File: web/src/components/TickerTable/index.tsx [hunk 3]
```diff
@@ -31,7 +35,7 @@ export const TickerTable: React.FC<Props> = ({
                     </td>
                     <td>
                         <span>
-                            <Decimal fixed={market.amount_precision} thousSep=",">
+                            <Decimal fixed={!isFixedPrecision ? market.amount_precision : market.price_precision} thousSep=",">
```
Comment (josadcha) [inline web/src/components/TickerTable/index.tsx:38]: Market.last is market last price. For displaying price we use market.price_precision.

File: web/src/components/TickerTable/index.tsx [hunk 4]
```diff
@@ -41,21 +45,21 @@ export const TickerTable: React.FC<Props> = ({
...
```
Comment (josadcha) [inline web/src/components/TickerTable/index.tsx:48]: same for market.high, it's the highest price within 24h, so again for price we use price_precision.

File: web/src/components/TickerTable/index.tsx [hunk 5]
```diff
@@ -41,21 +45,21 @@ export const TickerTable: React.FC<Props> = ({
...
                                 {market.high}
                             </Decimal>
                         </span>
                     </td>
                     <td>
                         <span>
-                            <Decimal fixed={market.amount_precision} thousSep=",">
+                            <Decimal fixed={!isFixedPrecision ? market.amount_precision : market.price_precision} thousSep=",">
```
Comment (josadcha) [inline web/src/components/TickerTable/index.tsx:55]: same here

File: web/src/components/TickerTable/index.tsx [hunk 6]
```diff
@@ -41,21 +45,21 @@ export const TickerTable: React.FC<Props> = ({
...
                                 {market.low}
                             </Decimal>
                         </span>
                     </td>
                     <td>
                         <span>
-                            <Decimal fixed={market.amount_precision} thousSep=",">
+                            <Decimal fixed={!isFixedPrecision ? market.amount_precision : FIXED_PRECISION} thousSep=",">
```
Comment (josadcha) [inline web/src/components/TickerTable/index.tsx:62]: market.volume is a value in quote currency. that's why here we also put market.price_precision

File: web/src/containers/MarketsTable/index.tsx [hunk 7]
```diff
@@ -101,6 +101,7 @@ const MarketsTableComponent = props => {
             markets={filteredMarkets}
             redirectToTrading={handleRedirectToTrading}
             setCurrentBidUnit={setCurrentBidUnit}
+            isFixedPrecision={props.isFixedPrecision}
```
Comment (josadcha) [inline web/src/containers/MarketsTable/index.tsx:104]: no need for this

File: web/src/mobile/screens/Landing/index.tsx [hunk 8]
```diff
@@ -4,7 +4,7 @@ import { MarketsTable } from '../../../containers';
 const LandingComponent: React.FC = () => {
     return (
         <div className="pg-landing-screen-mobile">
-            <MarketsTable />
+            <MarketsTable isFixedPrecision />
```
Comment (josadcha) [inline web/src/mobile/screens/Landing/index.tsx:7]: remove please this change

File: web/src/screens/LandingScreen/index.tsx [hunk 9]
```diff
@@ -94,7 +94,7 @@ class Landing extends React.Component<Props> {
                                     {this.translate('page.body.landing.marketInfo.title.button')}
                                 </Link>
                             </div>
-                            <MarketsTable />
+                            <MarketsTable isFixedPrecision />
```
Comment (josadcha) [inline web/src/screens/LandingScreen/index.tsx:97]: remove please this change

**Review by josadcha — CHANGES_REQUESTED** [review]

**Review by josadcha — APPROVED** [review]

**Review by akhlopiachyi — APPROVED** [review]

**Pull request merged**